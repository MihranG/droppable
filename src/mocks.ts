export const initialStateMock =  {
    "draggingID": -1,
    "items":  [
        {
            "isVisible": true,
            "value": 1,
        },
        {
            "isVisible": true,
            "value": 2,
        },
        {
            "isVisible": true,
            "value": 3,
        },
        {
            "isVisible": true,
            "value": 4,
        },
        {
            "isVisible": true,
            "value": 5,
        },
        {
            "isVisible": true,
            "value": 6,
        },
        {
            "isVisible": true,
            "value": 7,
        },
        {
            "isVisible": true,
            "value": 8,
        },
        {
            "isVisible": true,
            "value": 9,
        },
        {
            "isVisible": true,
            "value": 10,
        },
        {
            "isVisible": true,
            "value": 11,
        },
        {
            "isVisible": true,
            "value": 12,
        },
        {
            "isVisible": true,
            "value": 13,
        },
        {
            "isVisible": true,
            "value": 14,
        },
        {
            "isVisible": true,
            "value": 15,
        },
        {
            "isVisible": true,
            "value": 16,
        },
    ],
};


export const boxItemPropMock = {
    isVisible: true,
    value: 5
}