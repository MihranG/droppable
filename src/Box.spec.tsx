import React from 'react';
import ReactDOM from 'react-dom';
import {act, create,} from 'react-test-renderer';
import {Box} from "./Box";
import {boxItemPropMock} from "./mocks";


describe('Box component', () => {
    let container;
    jest.spyOn(Math, 'random').mockImplementation(() => 0.5);
    const onDragStartSpied = jest.fn((id) => () => 'onDragStart' + id);
    const onDragStopSpied = jest.fn(() => () => 'onDragStop');
    const onDropSpied = jest.fn((id) => () => 'onDrop' + id);

    beforeEach(() => {
        container = document.createElement('div');
        act(() => {
            ReactDOM.render(<Box
                initialItem={boxItemPropMock}
                index={6}
                onDrop={onDropSpied}
                onDragStart={onDragStartSpied}
                onDragStop={onDragStopSpied}
            />, container);
        })
    });

    afterEach(() => {
        ReactDOM.unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    describe('should be rendered', () => {
        let BoxComponent: any;
        beforeEach(() => {
            act(() => {
                BoxComponent = create(<Box
                    initialItem={boxItemPropMock}
                    index={6}
                    onDrop={jest.fn((id) => () => 'onDrop' + id)}
                    onDragStart={jest.fn((id) => () => 'onDragStart' + id)}
                    onDragStop={jest.fn(() => () => 'onDragStop')}
                />, container);
            });
        });

        it('should match snapshots ', () => {
            expect(BoxComponent.toJSON()).toMatchSnapshot();
        });

        it('should have appropriate event handlers', (done) => {
            // todo need to figure out why event dispatchers are not working correctly

            const fifthBox = container.querySelector('[data-testid="box-5"]');

            // ReactTestUtils.Simulate.dragStart(BoxComponent);
            fifthBox.dispatchEvent(new Event('dragStart'));
            fifthBox.dispatchEvent(new Event('dragEnd'));
            fifthBox.dispatchEvent(new Event('drop'));
            setTimeout(() => {
                expect(onDragStartSpied).toHaveBeenCalled();
                expect(onDragStopSpied).toHaveBeenCalled();
                expect(onDropSpied).toHaveBeenCalled();
            }, 0);
            done()
        });
    })
});