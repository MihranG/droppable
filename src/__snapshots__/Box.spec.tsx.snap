// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component should be rendered should match snapshots  1`] = `
<li
  className="css-12x2kaz"
  data-testid="box-5"
  draggable={true}
  onDragEnd={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
>
  <span>
     
    5
  </span>
</li>
`;
